### PR TITLE
Checker overhaul: Compile tests with glslang, fix tests, compile golden tests

### PR DESCRIPTION
**Pull request by therontarigo**
- Checker:
  - Fix 9 tests/unit/ compilation failures caused by empty minifier output
  - Don't fail on compiler warnings
  - Use glslang as the shader compiler by default
  - Support shader stages and non-compiled unit tests
  - Compile golden tests as specified in compile.txt
  - Support Shadertoys and HLSL
- Tests:
  - Fix compilation failures caused by use of reserved keywords
  - Fix compilation failures by adding appropriate #version directives
  - Misc. fixes where the test files were not valid shaders
- `-linux` project files: Use .NET Framework v4.8 for OpenTK.dll compatibility

### CODE REVIEW

**Comment by therontarigo** [comment]
The following failures are known:

```
tests/out/real/to_the_road_of_ribbon.minind.frag
  Minifier bug: same name used for a variable and for a function
  (Nvidia) 1(28) : error C1105: cannot call a non-function
  (Nvidia) 1(34) : error C1105: cannot call a non-function

tests/out/real/audio-flight-v2.minind.frag
tests/out/real/orchard.minind.frag
  Glslang bug?  Does the spec allow omitting the space?
  (Glslang) WARNING: 1:11: '#define' : missing space after macro name
  (Glslang) ERROR: 1:178: 'MINDIST' : undeclared identifier

tests/unit/inline.frag
  Minifier bug - I haven't checked what went wrong here
  (Glslang) ERROR: stdin:2: 'c' : undeclared identifier
  (Mesa) 0:2(740): error: `c' undeclared
  (Mesa) 0:2(733): error: could not implicitly convert return value to float, in
 function `D'
  (Nvidia) 0(2) : error C1016: expression type incompatible with function return
 type

tests/unit/shadowing.frag
  Minifier bug: "return i+++++i+i"
  (Glslang) ERROR: 1:1: '++' :  l-value required
  (Mesa) 1:1(30): error: assignment to post-increment operation
  (Nvidia) 1(1) : error C1070: increment/decrement of non-lvalue
```

**Comment by therontarigo** [comment]
Continuous Integration is attempting to run `Checker\bin\Debug\Checker.exe --skip-glsl-compile` but that option no longer exists.  I think this is set somewhere external to the repository.

**Comment by laurentlb** [comment]
Thanks a lot!

New build with `Checker.exe --compile-golden`: https://ci.appveyor.com/project/laurentlb/shader-minifier-r5gmt/builds/49868383

**Comment by therontarigo** [comment]
Should the failing tests be disabled for now until the minifier bugs can be individually fixed?

**Comment by laurentlb** [comment]
Yes, please. We can fix the issues separately, but I'd like the CI to be green to merge the change.

File: tests/unit/inline.frag [hunk 1]
```diff
@@ -115,7 +115,7 @@ void lvalues() {
 }
 
 uniform int time;
-in int sync;
+flat in int sync;
```
[the remark this answers is not in the document]
Comment (therontarigo) [reply tests/unit/inline.frag:118]: Any reason *not* to test `flat in` here?

[the remark this answers is not in the document]
Comment (laurentlb) [reply tests/unit/inline.frag:118]: no, I misread and thought it was a typo :)

**Comment by laurentlb** [comment]
Merged

Thanks a lot! ❤️

**Comment by eldritchconundrum** [comment]
Shouldn't `layout.*` replace `precision.frag`?

**Comment by therontarigo** [comment]
Right, I should have deleted everything from `precision.frag` that's now duplicated in `layout.*` - but it's good to keep a test of `precision` qualifier as well.  I didn't know if there is some reason both needed to be tested in the same file, it looks like not.

**Comment by therontarigo** [comment]
Oops, I've reset the auto-merge.  That's all of the failing tests now filed as issues and identified in comments.

**Review by laurentlb — APPROVED** [review]

**Pull request merged**